pkg/webhook/risingwave_validating_webhook.go

```diff
@@ -178,7 +178,7 @@ func (v *RisingWaveValidatingWebhook) validateMetaStoreAndStateStore(path *field
 			}
 		} else {
 			// AWS S3.
-			if !pointer.BoolDeref(stateStore.S3.UseServiceAccount, false) == (stateStore.S3.RisingWaveS3Credentials.SecretName == "") {
+			if !pointer.BoolDeref(stateStore.S3.UseServiceAccount, false) && stateStore.S3.RisingWaveS3Credentials.SecretName == "" {
 				fieldErrs = append(fieldErrs, field.Invalid(path.Child("stateStore", "s3", "credentials"), stateStore.S3.SecretName, "either secretName or useServiceAccount must be specified"))
 			}
```

Comment:
To double check: looks like the GCS one has the same inconsistency?
https://github.com/risingwavelabs/risingwave-operator/blob/94405ee8b38277623e3d0b5e6321cda9ae0c3c96/apis/risingwave/v1alpha1/risingwave_state_store_backend.go#L129-L131

https://github.com/risingwavelabs/risingwave-operator/blob/94405ee8b38277623e3d0b5e6321cda9ae0c3c96/pkg/webhook/risingwave_validating_webhook.go#L187-L190

Reply:
Fixed!